content/techniques/configuration.md

```diff
@@ -153,6 +153,25 @@ As shown above, use the `configService.get()` method to get a simple environment
 const dbHost = this.configService.get<string>('database.host', 'localhost');
 ```
 
+`ConfigService` has an optional generic to help prevent accessing a config property that does not exist.
+
+```typescript
+interface EnvironmentVariables {
+  PORT: number;
+  TIMEOUT: string;
+}
+
+constructor(private configService: ConfigService<EnvironmentVariables>) {
+  // This is valid
+  const port = this.configService.get<number>('PORT');
+  
+  // This is invalid as URL is not a property on the EnvironmentVariables interface
+  const url = this.configService.get<string>('URL');
+}
+```
+
+> warning **Notice** If you have nested properties in your config like in the `database.host` example above then your interface will need to have a `'database.host': string;` property, otherwise a TypeScript error will be thrownx1.
```

Comment:
```suggestion
> warning **Notice** If you have nested properties in your config, like in the `database.host` example above,  the interface must have a matching `'database.host': string;` property.  Otherwise a TypeScript error will be thrown.
```

Reply:
Left a few small comments.  Otherwise looks great!
@kamilmysliwiec: review completed

Reply:
great thank you @johnbiundo !